dist/index.js

```diff
@@ -0,0 +1,166 @@
+  tasks[index].completed = !tasks[index].completed; //change the status
+  if (tasks[index].completed) {
+    todoItemElems[index].classList.add('checked');
+  } else {
+    todoItemElems[index].classList.remove('checked');
+  }
+  updateLocal();
+  addToHTML();
+};
+/* eslint-anable */
+addTasksBtn.addEventListener('click', () => { // by clicking the button
+  tasks.push(new Task(deskTaskInput.value)); // add new task
+  updateLocal(); // save to Local Storage
+  addToHTML(); // call function wich add new task to HTML template
+  deskTaskInput.value = ''; // clear the input after clicking "Add" button
+
+});
+
+const deleteTask = index => { // function for delete button
+  todoItemElems[index].classList.add('deletion');
+  setTimeout(() => { // create a pause for animation when delete
+    tasks.splice(index, 1); // splice delete the index
+    updateLocal();
+    addToHTML();
+  }, 500);
+};
+class DND {
+  constructor() {
+    this.prevRow;
+  }
+
+  setEventListeners() {
+    let listItems = document.querySelectorAll('li');
+  
+    listItems.forEach((listItem) => {
+  
+      listItem.addEventListener('dragstart', (e) => this.start(e));
+      listItem.addEventListener('dragover', (e) => this.over(e));
+      listItem.addEventListener('drop', (e) => this.drop(e));  
+    });
+  }
+
+  start(e) {
+    this.prevRow = e.target;
+
+    let HTMLContent = e.target.innerHTML;
+    let checkboxStatus = e.target.querySelector('input').checked;
+
+    e.dataTransfer.setData('html-content', HTMLContent);
+    e.dataTransfer.setData('checkbox-status', checkboxStatus);
+  }
+
+  over(e) {
+    let currRow;
+
+    if (e.target.parentNode.tagName === 'LI') currRow = e.target.parentNode;
+    else if (e.target.parentNode.tagName === 'DIV') currRow = e.target.parentNode.parentNode;
+    else currRow = e.target;
+
+    e.preventDefault();
+
+    if (this.prevRow !== currRow) {
+      this.prevRow.innerHTML = currRow.innerHTML;
+      this.prevRow.querySelector('input').checked = currRow.querySelector('input').checked;
+      currRow.innerHTML = '';
+      this.prevRow = currRow;
+    }
+  }
+
+  drop(e) {
+    const HTMLContent = e.dataTransfer.getData('html-content');
+    const checkboxStatus = e.dataTransfer.getData('checkbox-status');
+
+    e.target.innerHTML = HTMLContent;
+    e.target.querySelector('input').checked = (checkboxStatus === 'true');
+  }
+
+}
+
+function init() {
+  let dnd = new DND(); 
+  dnd.setEventListeners();  
+}
+
+let windowLoad = new Promise(function(resolve) {
+  window.addEventListener('load', resolve);
+});
+
+windowLoad.then(
+  function(result) {
+    init();
+  }
+);
+
+function updateStatus(event, status) {
+  event.target.nextElementSibling.classList.toggle('completed');
+  status[event.target.dataset.id].completed = event.target.checked;
+  return status;
+}
+function editTodo(event, editable) {
+  editable[event.target.dataset.index].description = event.target.value;
+  return editable;
+}
```

Comment:
- [ ] As per the learning objective, We have to use proper ES6 syntax. Kindly try to use the arrow function as per ES6 guidelines.

Reply:
fixed